### ISSUE TITLE
Parse error for message chunk could cause client to crash

### ISSUE THREAD

**Issue by isaacgr** [isaacgr/jaysonic#25]
If a parse error is called, it uses the serving_message_id. The pending calls could have already resolved this ID, so there is nothing to call reject() against. This results in something like the below traceback.

```sh
TypeError: Cannot read property 'reject' of undefined
[1]     at TCPClient.handleError (/mnt/c/Users/irowell/Documents/node_modules/jaysonic/lib/client/index.js:440:39)
[1]     at TCPClient.verifyData (/mnt/c/Users/irowell/Documents/node_modules/jaysonic/lib/client/index.js:358:24)
[1]     at Socket.<anonymous> (/mnt/c/Users/irowell/Documents//node_modules/jaysonic/lib/client/index.js:404:24)
[1]     at Socket.emit (events.js:189:13)
[1]     at addChunk (_stream_readable.js:284:12)
[1]     at readableAddChunk (_stream_readable.js:261:13)
[1]     at Socket.Readable.push (_stream_readable.js:220:10)
[1]     at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
```

**Comment by isaacgr** [comment]
This appeared to be caused by the server sending back data split between two streams. So the other messages in the stream get parsed, but since the incomplete stream is invalid json it sends the error.

**Comment by isaacgr** [comment]
Current solution is to try and reject the serving message if there is a parse error, and if there is no pending call then just log the error output.

**Comment by isaacgr** [comment]
Fixed in #26
